File: 1313. Decompress Run-Length Encoded List/solution.java
class Solution {
  public int[] decompressRLElist(int[] nums) {
    List<Integer> result = new ArrayList<Integer>();
    for (int index = 0; index < nums.length - 1; index += 2){
      int frequency = nums[index];
      int value = nums[index + 1];
      for (int amount = 0; amount < frequency; amount++){
        result.add(value);
      }
    }
    
    int[] resultingArray = new int[result.size()];
    for (int index = 0; index < result.size(); index++){
      resultingArray[index] = result.get(index);
    }

    return resultingArray;
  }
}

File: 1313. Decompress Run-Length Encoded List/solution.cpp
class Solution {
public:
  vector<int> decompressRLElist(vector<int>& nums) {
    vector<int> result;
    for (int index = 0; index < nums.size() - 1; index += 2){
      int frequency = nums[index];
      int value = nums[index + 1];
      for (int amount = 0; amount < frequency; amount++){
        result.push_back(value);
      }
    }

    return result;
  }
};